File: lamport_clock/lamport_clock.py

class LamportClock:
    def __init__(self):
        self.clock = 0

    def tick(self):
        self.clock += 1

    def update_clock(self, received_clock):
        if received_clock > self.clock:
            self.clock = received_clock + 1
        else:
            self.clock += 1

    def get_clock(self):
        return self.clock

    def print_clock(self):
        print(f"Clock: {self.clock}")

File: old_go_version/lamport_clock/lamport_clock.go
package lamport_clock

import ("fmt"; "sync")

type LamportClock struct {
	clock int
	mutex sync.Mutex
}

func NewLamportClock() *LamportClock {
	return &LamportClock{
		clock: 0,
	}
}

func (lc *LamportClock) Tick() {
	lc.mutex.Lock()
	defer lc.mutex.Unlock()

	lc.clock++
}

func (lc *LamportClock) UpdateClock(receivedClock int) {
	lc.mutex.Lock()
	defer lc.mutex.Unlock()

	if receivedClock > lc.clock {
		lc.clock = receivedClock + 1
	} else {
		lc.clock++
	}
}

func (lc *LamportClock) GetClock() int {
	lc.mutex.Lock()
	defer lc.mutex.Unlock()

	return lc.clock
}

func (lc *LamportClock) PrintClock() {
	fmt.Printf("Clock: %d\n", lc.clock)
}